pyro/distributions/mixture.py

```diff
@@ -102,14 +102,16 @@ def expand(self, batch_shape):
 
     def sample(self, sample_shape=torch.Size()):
         mask = self.mask.expand(sample_shape + self.batch_shape) if sample_shape else self.mask
-        result = self.component0.sample(sample_shape)
-        result[mask] = self.component1.sample(sample_shape)[mask]
+        result = torch.where(mask,
```

Comment:
I think you would need to reshape `mask` here: `mask.reshape(mask.shape + (1,) * len(self.event_shape))`

Reply:
Good catch, it looks like this is causing `tests/distributions/test_mixture.py::test_masked_mixture_multivariate` to fail. @ordabayevy can you see if this fixes the test?

Reply:
@fritzo yeah, I am looking into it

Reply:
@fritzo where can I find instructions on how to run the tests? I looked [here](https://github.com/pyro-ppl/pyro/tree/dev/tests) but couldn't build pytorch binaries in docker. Some help would be greatly appreciated since I am new to many of these things.

Reply:
I've run the test locally as `pytest -q test_mixture.py` and it passed it with @fehiepsi suggestion.